tests/src/test/java/org/neo4j/graphalgo/core/CypherExporter.java

```diff
@@ -24,15 +25,27 @@ static void export(PrintWriter out, GraphDatabaseService db) {
                                     node.getRelationships(Direction.OUTGOING)
                             )
                     )
-                    .map(rel ->
-                            "  (n" + rel.getStartNode().getId() + ")"
-                                    + "-[:TYPE]->"
-                                    + "(n" + rel.getEndNode().getId() + ")"
-                    )
+                    .map(CypherExporter::rel)
                     .collect(Collectors.joining(",\n", "CREATE\n", "\n"))
             );
             tx.success();
         }
         out.flush();
     }
+
+    private static String rel(Relationship rel) {
+        String props = rel.getAllProperties()
+                .entrySet()
+                .stream()
+                .map(e -> e.getKey() + ":" + e.getValue())
```

Comment:
Currently, we only use this for tests that generate a random graph in cypher so that we can "export" the graph and dump it in case of test failures, if we want to reproduce the test with the graph that has failed. Since we only generate the property `weight` with doubles, we're good here.
I'll add a note to the class that it's not intended to be used by anyone else besides the test.